### PR TITLE
qemu: bump to 7.1, add products for user-mode emulation binaries

### PR DESCRIPTION
**Pull request by maleadt**
cc @simeonschaub

### CODE REVIEW

**Comment by maleadt** [comment]
The error:

```
[15:07:07] BinaryBuilder: Cannot force an architecture via -march
[15:07:07] make[2]: *** [Makefile:41: linuxboot_dma.o] Error 1
```

**Comment by giordano** [comment]
That'd be https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/778

File: Q/Qemu/build_tarballs.jl [hunk 1]
```diff
@@ -57,38 +69,85 @@ platforms = [
 ]
 platforms = expand_cxxstring_abis(platforms)
 
+# some platforms need a newer glibc, because the default one is too old
+glibc_platforms = filter(platforms) do p
+    libc(p) == "glibc" && proc_family(p) in ["intel", "power"]
```
Comment (simeonschaub) [inline Q/Qemu/build_tarballs.jl:74]: Did you want to run this filter over all available platforms? Right now it's only x64 and i686 Linux anyways.

Comment (maleadt) [reply Q/Qemu/build_tarballs.jl:74]: Just to keep the code generic -- the same pattern exists in other recipes where glibc is upgraded (crun, systemd).

**Pull request merged**